unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs

```diff
@@ -50,19 +50,20 @@ public void Dispose()
 
         public void Update()
         {
-            string currentSceneId = state.worldState.GetCurrentSceneId();
-            bool sceneChanged = state.lastSceneId != currentSceneId;
-            state.lastSceneId = currentSceneId;
+            int currentSceneNumber = state.worldState.GetCurrentSceneNumber();
+            bool sceneChanged = state.lastSceneNumber != currentSceneNumber;
+            state.lastSceneNumber = currentSceneNumber;
 
             ApplyParenting(state.uiDocument, state.internalUiContainerComponent);
 
             // clear UI if scene changed
             if (sceneChanged && !state.isPendingSceneUI)
             {
                 ClearCurrentSceneUI(state.uiDocument);
-                state.isPendingSceneUI = !string.IsNullOrEmpty(currentSceneId);
+                state.isPendingSceneUI = currentSceneNumber > 0;
```

Comment:
not sure about this check, could we have negative numbers for real scenes? i've seen the usage of `newGuid().GetHashCode()`, perhaps those are not real calculated scene ids?

Reply:
Kernel will never provide a scene number that's 0 or less, if that happened, it would be an invalid scene number. As stated in this PRs description all valid scene numbers are now `> 0` (no more scene id null checks).

Reply:
Also, the HashCode of a class has nothing to do with scenes scene number

Reply:
ok!